hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -103,6 +104,9 @@ public BaseSparkCommitActionExecutor(HoodieEngineContext context,
     if (isWorkloadProfileNeeded()) {
```

Comment:
what happens if workload profile is not needed? Is there a better place to do this validation? Can we do this after tagLocation is done? If any of the records have location to files in pending clustering, we can throw error.

Reply:
it is make sense.